metadoc-js/src/main/scala/metadoc/MetadocApp.scala

```diff
@@ -125,7 +129,30 @@ object MetadocApp {
     )
   }
 
-  def openEditor(editorService: MetadocEditorService)(
+  def registerWorkspaceFiles(
+      editor: IStandaloneCodeEditor,
+      workspace: Workspace
+  ): Unit = {
+    workspace.filenames.foreach { file =>
+      editor.addAction(new IActionDescriptor {
+        override var id = file
+        override var label = file
+        override def run(
+            editor: ICommonCodeEditor
+        ): monaco.Promise[Unit] = {
+          dom.window.location.hash = "#/" + file
```

Comment:
This messes up the history, I need to press "back" twice to switch files. Once to remove line+column and second to switch files. Is there a better way? I tried `updateTitle` and `updateHistory` but they didn't change the open buffer.

Reply:
This should use `openEditor(editorService)(state)` I think. It could be that `updateHistory` should be called inside `openEditor` so we only update history when `editorService.open` returns.